src/anyio/_backends/_asyncio.py

```diff
@@ -1762,10 +1765,14 @@ async def acquire(self) -> None:
         self._waiters.remove(item)
 
     def acquire_nowait(self) -> None:
+        task = cast(asyncio.Task, current_task())
         if self._owner_task is None and not self._waiters:
-            self._owner_task = current_task()
+            self._owner_task = task
             return
 
+        if self._owner_task == task:
```

Comment:
Why not use is?

```suggestion
        if self._owner_task is task:
```

Should slightly faster?